### PR TITLE
[web] Process keydown and keyup keys for identified keys from virtual keyboard

### PR DESCRIPTION
**Pull request by Schahen**
This fixes some issues with  pressing some virtual keys that don't change the input value - like navigation with arrows.
Also, this will make finally possible to switch to KeybardEvent::key-backed toComposeEvent conversion

### CODE REVIEW

File: compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/platform/BackingTextArea.kt [hunk 1]
```diff
@@ -41,10 +40,19 @@ internal class BackingTextArea(
     private val imeOptions: ImeOptions,
     private val onEditCommand: (List<EditCommand>) -> Unit,
     private val onImeActionPerformed: (ImeAction) -> Unit,
-    private val sendKey: (evt: KeyEvent) -> Unit
+    private val processKeyboardEvent: (KeyboardEvent) -> Unit
 ) {
     private val textArea: HTMLTextAreaElement = createHtmlInput()
 
+    fun processIdentifiedEvent(evt: Event) {
```
Comment (MatkovIvan) [inline compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/platform/BackingTextArea.kt]: Might be private

Comment (Schahen) [reply compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/platform/BackingTextArea.kt]: agreed

**Review by eymar — APPROVED** [review]
I'm thinking about a test which can be added here:

- the test would send a touch event to make our internal state (keyboardModeState) switch to  `KeyboardModeState.Virtual` (a tap on the TextField)
- it would send Keyboard events into the textarea
- it would check that Modifier.onKeyEvent receives the events

What do you think?

File: compose/ui/ui/src/webTest/kotlin/androidx/compose/ui/SelectionContainerTests.kt [hunk 2]
```diff
@@ -52,7 +52,7 @@ class SelectionContainerTests : OnCanvasTests {
     fun setup() {
         // Because AfterTest is fixed only in kotlin 2.0
         // https://youtrack.jetbrains.com/issue/KT-61888
-        document.getElementById(canvasId)?.remove()
+        commonAfterTest()
```
Comment (eymar) [inline compose/ui/ui/src/webTest/kotlin/androidx/compose/ui/SelectionContainerTests.kt:55]: The tests here failed on CI. 
Could this change be a reason of the test failure?

Comment (eymar) [reply compose/ui/ui/src/webTest/kotlin/androidx/compose/ui/SelectionContainerTests.kt:55]: although locally all the tests pass. weird :(

Comment (eymar) [reply compose/ui/ui/src/webTest/kotlin/androidx/compose/ui/SelectionContainerTests.kt:55]: Tried to rerun the CI, but same failure.

for some reason I think `commonAfterTest` might be a reason.

Comment (Schahen) [reply compose/ui/ui/src/webTest/kotlin/androidx/compose/ui/SelectionContainerTests.kt:55]: let me just return everything as it was and see how it goes
the reason might be parallelization or whatever

**Pull request merged**